bindings/experimental/distrdf/python/DistRDF/Backends/Dask/Backend.py

```diff
@@ -93,8 +111,8 @@ def optimize_npartitions(self) -> int:
         either retrieved if known or inferred from the user-provided cluster
         specification.
         """
-        # We dispatch on the type of the cluster, but the API to retrieve the
-        # number of available cores belongs to the client
+        # Depending on the cluster type we may need to use it, more generally
+        # the API to retrieve the number of cores is accessible from the client.
```

Comment:
I had to re-read this sentence, could not parse it on the first try -- maybe it's only me :)

I wonder if we even need to explain the inner workings of `get_total_cores` here though. We call a function called `get_total_cores`, it's obvious what it returns, all good from the point of view of the caller?

Reply:
Indeed, the comment is maybe unnecessary and can lead to confusion. The docstrings in the implementation functions already describe the different cases they are addressing